src/main/java/bio/terra/service/filedata/flight/ingest/IngestDriverStep.java

```diff
@@ -311,6 +315,20 @@ private LoadCandidates getLoadCandidates(FlightContext context, UUID loadId, int
     return candidates;
   }
 
+  /** Add existing `key`-val pair from `context`'s input parameters to `flightMap`. */
+  @VisibleForTesting
+  <T> void propagateContextToFlightMap(
+      FlightContext context, FlightMap flightMap, String key, Class<T> paramType) {
+    Optional<T> maybeParam;
+    try {
+      maybeParam = Optional.ofNullable(context.getInputParameters().get(key, paramType));
+    } catch (Exception ex) {
+      logger.error("Unable to deserialize context input parameter value", ex);
+      maybeParam = Optional.empty();
+    }
+    maybeParam.ifPresent(param -> flightMap.put(key, param));
```

Comment:
Personally I don't think `Optional` adds value here over comparing `!= null`. Not using `maybeParam` is less code too, I think.
```suggestion
    try {
      T param = context.getInputParameters().get(key, paramType);
      if (param != null) {
        flightMap.put(key, param);
      }
    } catch (Exception ex) {
      logger.error("Unable to deserialize context input parameter value", ex);
    }
```

Reply:
Fair.  The `Optional` added more value before I introduced the try-catch.